crates/ra_project_model/src/lib.rs

```diff
@@ -232,10 +234,28 @@ impl ProjectWorkspace {
                                 Env::default(),
                             );
                             names.insert(crate_id, pkg.name(&cargo).to_string());
-                            if tgt.kind(&cargo) == TargetKind::Lib {
-                                lib_tgt = Some(crate_id);
-                                pkg_to_lib_crate.insert(pkg, crate_id);
+                            match tgt.kind(&cargo) {
+                                TargetKind::Lib => {
+                                    lib_tgt = Some(crate_id);
+                                    pkg_to_lib_crate.insert(pkg, crate_id);
+                                }
+                                TargetKind::ProcMacro => {
```

Comment:
Hm, I think to hanle this case, we need to add a new field to `target`, `crate_type`. Or maybe just a `is_proc_macro: bool`

Reply:
The boolean field seems like the nicest solution for now, as only `proc-macro` requires special handling.